tests/components/hisense_aehw4a1/test_init.py

```diff
@@ -0,0 +1,81 @@
+"""Tests for the Hisense AEH-W4A1 init file."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components import hisense_aehw4a1
+from homeassistant.setup import async_setup_component
+
+from tests.common import mock_coro
+
+
+async def test_creating_entry_sets_up_climate_discovery(hass):
+    """Test setting up Hisense AEH-W4A1 loads the climate component."""
+    with patch(
+        "homeassistant.components.hisense_aehw4a1.climate.async_setup_entry",
+        return_value=mock_coro(True),
+    ) as mock_setup:
+        result = await hass.config_entries.flow.async_init(
+            hisense_aehw4a1.DOMAIN, context={"source": config_entries.SOURCE_DISCOVERY}
+        )
+
+        # Confirmation form
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+        result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+
+        await hass.async_block_till_done()
+
+    assert len(mock_setup.mock_calls) == 1
+
+
+async def test_configuring_hisense_w4a1_create_entry(hass):
+    """Test that specifying config will create an entry."""
+    with patch(
+        "homeassistant.components.hisense_aehw4a1.config_flow.AehW4a1.check",
+        return_value=mock_coro(True),
+    ):
+        with patch(
+            "homeassistant.components.hisense_aehw4a1.async_setup_entry",
+            return_value=mock_coro(True),
+        ) as mock_setup:
+            await async_setup_component(
+                hass,
+                hisense_aehw4a1.DOMAIN,
+                {"hisense_aehw4a1": {"ip_address": ["1.2.3.4"]}},
+            )
+            await hass.async_block_till_done()
+
+    assert len(mock_setup.mock_calls) == 1
+
+
+async def test_configuring_hisense_w4a1_not_create_entry_for_device_not_found(hass):
+    """Test that specifying config will not create an entry."""
+    with patch(
+        "homeassistant.components.hisense_aehw4a1.config_flow.AehW4a1.check",
+        return_value=mock_coro(exception="ConnectionError"),
```

Comment:
This shouldn't return a coroutine, right? `check` is just a normal function.

If we want to raise an exception we should set a side_effect.

Reply:
Ok, thanks!